kDriveCore/DI/FactoryService.swift

```diff
@@ -94,6 +94,9 @@ public enum FactoryService {
             Factory(type: TokenStore.self) { _, _ in
                 TokenStore()
             },
+            Factory(type: DownloadQueue.self) { _, _ in
```

Comment:
I understand you did just like the existing UploadQueue, but here it is simpler. You can keep it tidy with only registering a `DownloadQueueable` in the DI 👍 

Please also do this for UT / IT tests 🙏 

```suggestion
            Factory(type: DownloadQueueable.self) { _, _ in
```